### ISSUE TITLE
Allow users to specify a custom build.gradle when using standalone docker image

### ISSUE THREAD

**Issue by jaredculp** [spring-cloud/spring-cloud-contract#1672]
**Is your feature request related to a problem? Please describe.**

From a conversation on [gitter](https://gitter.im/spring-cloud/spring-cloud-contract?at=60b928f384a2f633a78bf566). My team has built contracts which we would like to validate against a nodejs based producer to run generated tests.

When using the docker image for contract testing, there is no way to customize the behavior of the Gradle build. Specifically:
* How to use static imports
* How to load custom jars onto the class path for things like body matchers
* How to configure proxy configuration

**Describe the solution you'd like**

Either environment variables to set the above or the ability to use a custom `build.gradle` file.

**Describe alternatives you've considered**

Creating a new base image from the spring-cloud-contract image.

**Additional context**

@marcingrzejszczak asked me to file this ticket.

**Comment by bastiandoetsch** [comment]
@jaredculp have you tried just mapping it into the container, using the -v option? E.g.

`docker run -v $PWD/yourCustomBuildGradle:/home/scc/spring-cloud-contract/build.gradle [...]` on current 3.0.x & main branch.

**Comment by marcingrzejszczak** [comment]
I think that the easiest way would to do it would be to use `apply from` in the current gradle file and point to a predefined path in the mounted volume (assuming that the file exists). WDYT?

**Comment by bastiandoetsch** [comment]
Would be an option, but I feel it would be easier to just pass a customized `build.gradle`. Adding `apply from:` and a custom directory just adds to complexity without really offering much more benefits. I'd rather update the documentation to have an example on how to mount a custom `build.gradle`.

**Comment by marcingrzejszczak** [comment]
If this `docker run -v $PWD/yourCustomBuildGradle:/home/scc/spring-cloud-contract/build.gradle [...]` is an equivalent then I'm all for it.

**Comment by bastiandoetsch** [comment]
Yeah, it just substitutes the `build.gradle` in the docker container with the local one left of the `:` at runtime.

On Aug 10, 2021, at 12:05, Marcin Grzejszczak ***@***.******@***.***>> wrote:



If this docker run -v $PWD/yourCustomBuildGradle:/home/scc/spring-cloud-contract/build.gradle [...] is an equivalent then I'm all for it.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fspring-cloud%2Fspring-cloud-contract%2Fissues%2F1672%23issuecomment-895900073&data=04%7C01%7Cbdoetsch%40vmware.com%7Cb2be7ae73ed546fa30bd08d95be650e7%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C637641867033750406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=eOpB0CovfGUSj%2BBMqe3GbNHkHCqpXHLEbn1KZWZhC8s%3D&reserved=0>, or unsubscribe<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAEZXT2LP5BHS4NLNWWHBSJTT4D2UZANCNFSM46ME3OXQ&data=04%7C01%7Cbdoetsch%40vmware.com%7Cb2be7ae73ed546fa30bd08d95be650e7%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C637641867033760363%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=4FMC3mG2PsBh4CvH8Q1v5eABKxQWMGQ%2Btr9zr2ZZFjM%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cbdoetsch%40vmware.com%7Cb2be7ae73ed546fa30bd08d95be650e7%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C637641867033760363%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=n0hBP%2ByE5L8m%2FVNVsVMFbpYX9FnXhyRcLZ%2BJ%2FQUppvg%3D&reserved=0> or Android<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7Cbdoetsch%40vmware.com%7Cb2be7ae73ed546fa30bd08d95be650e7%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C637641867033760363%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=rcAWSKNVSMLzWGRKXjhSu4WINI9hn%2FINfip3TnttsmY%3D&reserved=0>.

**Comment by bastiandoetsch** [comment]
I'll just add the description in `docker/spring-cloud-contract-docker/project/README.adoc` then.

**Comment by marcingrzejszczak** [comment]
And please add it to the docs under the docs module. There's an adoc file for the docker setup there
